prow/gerrit/reporter/reporter.go

```diff
@@ -62,33 +65,83 @@ func (c *Client) ShouldReport(pj *v1.ProwJob) bool {
 	}
 
 	// has gerrit metadata (scheduled by gerrit adapter)
-	return pj.ObjectMeta.Annotations["gerrit-id"] != "" &&
-		pj.ObjectMeta.Annotations["gerrit-instance"] != "" &&
-		pj.ObjectMeta.Labels["gerrit-revision"] != ""
+	if pj.ObjectMeta.Annotations[gerrit.GerritID] == "" ||
+		pj.ObjectMeta.Annotations[gerrit.GerritInstance] == "" ||
+		pj.ObjectMeta.Labels[gerrit.GerritRevision] == "" {
+		return false
+	}
+
+	// has other jobs on the same revision finished
+	selector := labels.Set{gerrit.GerritRevision: pj.ObjectMeta.Labels[gerrit.GerritRevision]}
+	pjs, err := c.lister.List(selector.AsSelector())
+	if err != nil {
+		logrus.WithError(err).Errorf("Cannot list prowjob with selector %v", selector)
+		return false
+	}
+
+	for _, pj := range pjs {
+		if pj.Status.State == v1.TriggeredState || pj.Status.State == v1.PendingState {
+			// not done yet
+			return false
+		}
+	}
+
+	return true
 }
 
 // Report will send the current prowjob status as a gerrit review
 func (c *Client) Report(pj *v1.ProwJob) error {
```

Comment:
It seems like Report() needs to take a gerrit revision, not a prowjob

Reply:
It will need the status from a prowjob at least, and this is an interface method :-)

Reply:
It does not use the status from the pj argument. It gets the gerrit revision from the pj object metadata, which it uses to list all the prow jobs that have this revision. It does nothing else with the pj.

The prowjob status is found from the listed prowjobs.